src/combinatorics/simple_graph/metric.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2022 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Vincent Beffara
+-/
+import combinatorics.simple_graph.connectivity
+import data.nat.lattice
+
+/-!
+# Graph metric
+
+This module defines the `simple_graph.dist` function, which takes
+pairs of vertices to the length of the shortest walk between them.
+
+## Main definitions
+
+- `simple_graph.dist` is the graph metric.
+
+## Todo
+
+- Provide an additional computable version of `simple_graph.dist`
```

Comment:
A directed graph version of this? You don't get symmetry but I think the function could still be useful

Reply:
Certainly a directed graph version, except that for the moment directed graphs do not exist yet. They will!